akka-cluster-typed/src/main/scala/akka/cluster/typed/internal/receptionist/ClusterReceptionist.scala

```diff
@@ -162,7 +170,7 @@ private[typed] object ClusterReceptionist extends ReceptionistBehaviorProvider {
        * Hack to allow multiple termination notifications per target
        * FIXME #26505: replace by simple map in our state
        */
-      def watchWith(ctx: ActorContext[Command], target: ActorRef[_], msg: InternalCommand): Unit = {
+      def watchWith(ctx: ActorContext[Command], target: ActorRef[_], msg: InternalCommand): ActorRef[Nothing] = {
```

Comment:
Does this add compiler errors for any pre-existing calls to the function expecting/handling `Unit`?